src/bio/SConscript

```diff
@@ -9,6 +9,7 @@ def scons():
     env.AppendUnique(LIBPATH=[Dir('.')])
 
     denv = env.Clone()
+    denv.AppendUnique(LINKFLAGS=["-Wl,--no-as-needed"])
```

Comment:
I'm bit confused, the "as-needed" option is a Ubuntu global setting or SPDK in Ubuntu setting? If it's a SPDK in Ubuntu setting, I think it's a SPDK defect and needs be fixed in SPDK? (this patch is just a workaround?)

Reply:
ah, I rechecked the commit comment, so the problem is rte_mempool_ring isn't linked because of 'as-needed' option is added when building DAOS in Ubuntu?

Is this 'as-needed' option being added by our build script or some sort Ubuntu default setting? Won't control plane suffer the similar problem?

Reply:
`--as-needed` is applied as a default linker option in Ubuntu global setting, we want the general behaviour to be `--no-as-needed` in all situations/all OSes. Some constructor functions in `rte_mempool_ring` don't get run unless we pass "-Wl,-no-as-needed". That is what I understand from Jim. We should probably add it to control-plane build script even though we are not seeing specific issues (we don't use thread_lib_init so maybe the specific constructor functions that were causing issues and missing when `--as-needed` are not called in the control-plane).

Reply:
I see. Thanks.